pkg/xds/generator/ingress_generator.go

```diff
@@ -160,7 +161,48 @@ func (_ IngressGenerator) destinations(
 		}
 	}
 
-	if len(policies[meshhttproute_api.MeshHTTPRouteType].GetItems()) > 0 {
+	addMeshHTTPRoutesDestinations(policies[meshhttproute_api.MeshHTTPRouteType],
+		destinations)
+
+	var backends []*mesh_proto.MeshGatewayRoute_Backend
+
+	for _, route := range ingressProxy.GatewayRoutes.Items {
+		for _, rule := range route.Spec.GetConf().GetHttp().GetRules() {
+			backends = append(backends, rule.Backends...)
+		}
+		for _, rule := range route.Spec.GetConf().GetTcp().GetRules() {
+			backends = append(backends, rule.Backends...)
+		}
+	}
+
+	for _, backend := range backends {
+		service := backend.Destination[mesh_proto.ServiceTag]
+		destinations[service] = append(destinations[service], backend.Destination)
+	}
+
+	for _, gateway := range ingressProxy.MeshGateways.Items {
+		for _, selector := range gateway.Selectors() {
+			service := selector.GetMatch()[mesh_proto.ServiceTag]
+			for _, listener := range gateway.Spec.GetConf().GetListeners() {
+				if !listener.CrossMesh {
+					continue
+				}
+				destinations[service] = append(
+					destinations[service],
+					tags.Tags(mesh_proto.Merge(selector.GetMatch(), gateway.Spec.GetTags(), listener.GetTags())),
+				)
+			}
+		}
+	}
+
+	return destinations
+}
+
+func addMeshHTTPRoutesDestinations(
+	policyResources core_model.ResourceList,
```

Comment:
nit: but probably this should already be a `[]*MeshHTTPRouteResource`

Reply:
Addressed